File: Thesis/middle-tier/app/services/journal_service.py
from app.utils.edge_function_caller import call_edge_function

def log_journal(token: str, content: str, date: str, moods: list[str]):
    return call_edge_function("logJournal", token, {
        "content": content,
        "date": date,
        "moods": moods
    })

def get_journal(token: str, date: str):
    return call_edge_function("getJournal", token, {"date": date}, method="GET")

def get_all_journals(token: str):
    return call_edge_function("getAllJournals", token, {}, method="GET")

def update_journal(token: str, journal_id: str, content: str, moods: list[str]):
    return call_edge_function("updateJournal", token, {
        "journalId": journal_id,
        "content": content,
        "moods": moods
    }, method="PUT")

def delete_journal(token: str, journal_id: str):
    return call_edge_function("deleteJournal", token, {
        "journalId": journal_id
    }, method="DELETE")

File: Thesis/frontend/services/journalService.ts
import api from './api';

export async function logJournal(content: string, date: string, moods: string[]) {
  console.log("📤 Sending logJournal payload:", {
    content,
    date,
    moods,
  });

  const res = await api.post("log-journal", {
    content,
    date,
    moods,
  });

  return res.data.data;
}


export const getJournal = async (date: string, token?:string) => {

   const res = await api.get(`/get-journal/${date}`,
    token ? { headers: { Authorization: token } } : {});

  return res.data.data.journal;
};

export const getAllJournals = async () => {
  const res = await api.get('/get-all-journals');
  return res.data.data.journals;
};

export async function updateJournal(id: string, content: string, moods: string[]) {
  console.log("📤 Sending updateJournal payload:", {
    id,
    content,
    moods,
  });

  const res = await api.put(`update-journal/${id}`, {
    content,
    moods,
  });

  return res.data.data;
}


export const deleteJournal = async (journal_id: string, token?: string) => {
  const res = await api.delete(
    `/delete-journal/${journal_id}`,
    token ? { headers: { Authorization: token } } : {}
  );
  return res.data.data;
};
